lib/src/license.dart

```diff
@@ -50,104 +51,22 @@ Future<LicenseFile> detectLicenseInFile(File file,
   return license ?? LicenseFile(relativePath, LicenseNames.unknown);
 }
 
+/// Returns an instance of [LicenseFile] if [originalContent] has contains a license(s)
+/// present in the [SPDX-corpus][1].
+///
+/// [1]: https://spdx.org/licenses/
 LicenseFile? detectLicenseInContent(String originalContent,
     {required String relativePath}) {
   var content = originalContent;
-  if (content.startsWith('// ')) {
-    content = content.split('\n').map((s) {
-      return s.startsWith('// ') ? s.substring(3) : s;
-    }).join('\n');
-  }
-  var stripped = _longTextPrepare(content);
-
-  String? version;
-  var versionMatch = _version.firstMatch(stripped);
-  if (versionMatch != null) {
-    version = versionMatch.group(1);
-    if (version!.isNotEmpty && !version.contains('.')) {
-      version += '.0';
-    }
-  }
+  final licenseResult = detectLicense(content, 0.85);
 
-  if (_mpl.hasMatch(stripped)) {
-    return LicenseFile(relativePath, LicenseNames.mpl, version: version);
-  }
-  if (_agpl.hasMatch(stripped) && !_useWithAgpl.hasMatch(stripped)) {
-    return LicenseFile(relativePath, LicenseNames.agpl, version: version);
-  }
-  if (_apache.hasMatch(stripped)) {
-    return LicenseFile(relativePath, LicenseNames.apache, version: version);
-  }
-  if (_lgpl.hasMatch(stripped) && !_useLgplInstead.hasMatch(stripped)) {
-    return LicenseFile(relativePath, LicenseNames.lgpl, version: version);
-  }
-  if (_gplLong.hasMatch(stripped)) {
-    return LicenseFile(relativePath, LicenseNames.gpl, version: version);
-  }
-  if (_gplShort.hasMatch(stripped)) {
-    return LicenseFile(relativePath, LicenseNames.gpl, version: version);
-  }
-  if (_mit.hasMatch(stripped) || _mitEmphasis.hasMatch(stripped)) {
-    return LicenseFile(relativePath, LicenseNames.mit, version: version);
-  }
-  if (_unlicense.hasMatch(stripped)) {
-    return LicenseFile(relativePath, LicenseNames.unlicense, version: version);
-  }
-
-  if (_bsdPreamble.hasMatch(stripped) && _bsdEmphasis.hasMatch(stripped)) {
-    return LicenseFile(relativePath, LicenseNames.bsd);
+  if (licenseResult.matches.isNotEmpty &&
+      licenseResult.unclaimedTokenPercentage <= 0.5) {
```

Comment:
Let's also reject all matches if the longest sequence of unclaimed tokens is longer than 50 or 100?

We can try different values can see how it affects our ability to detect licenses...